src/sync/sync_user.hpp

```diff
@@ -100,19 +115,32 @@ friend class SyncSession;
     std::string refresh_token() const;
     State state() const;
 
+    util::AtomicSharedPtr<SyncUserContext> binding_context;
+
     // Register a session to this user.
     // A registered session will be bound at the earliest opportunity: either
     // immediately, or upon the user becoming Active.
     // Note that this is called by the SyncManager, and should not be directly called.
     void register_session(std::shared_ptr<SyncSession>);
 
+    // Optionally set a context factory. If so, must be set before any sessions are created.
+    static void set_binding_context_factory(SyncUserContextFactory& factory)
+    {
+        s_binding_context_factory = &factory;
```

Comment:
This likely requires some form of synchronization since it may be set from a different thread than on which it is later read.